src/Module.cpp

```diff
@@ -319,9 +320,10 @@ void compile_multitarget(const std::string &fn_name,
     }
 
     std::vector<std::future<void>> futures;
-    // If we are running with HL_DEBUG_CODEGEN=1, choose a policy that enforces
+    // If we are running with HL_DEBUG_CODEGEN=1, use threads=1 to enforce
     // sequential execution, so that debug output won't be utterly incomprehensible
-    std::launch policy = debug::debug_level() > 0 ? std::launch::deferred : std::launch::async;
+    const size_t num_threads = (debug::debug_level() > 0) ? 1 : (targets.size() + 3);
```

Comment:
OK, I see. My thinking here was that since these tasks are heavily IO bound, tying it to number of CPUs didn't matter. That said, it probably doesn't matter much on any modern machine, and what I have written there is confusing. Will fix.